spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/SparkShufflingDataRewriter.java

```diff
@@ -122,11 +136,12 @@ private Dataset<Row> transformPlan(Dataset<Row> df, Function<LogicalPlan, Logica
   }
 
   private org.apache.iceberg.SortOrder outputSortOrder(List<FileScanTask> group) {
-    boolean includePartitionColumns = !group.get(0).spec().equals(table().spec());
+    PartitionSpec spec = table().specs().get(outputSpecId());
+    boolean includePartitionColumns = !group.get(0).spec().equals(spec);
```

Comment:
Let's rename this variable, since I think the original name is confusing here. Maybe just call it requiresRepartitioning